### PR TITLE
Enable batch in bnsd

### PR DESCRIPTION
**Pull request by ethanfrey**
Enabling batch in bnsd, as it is most likely needed in the governance contracts for token distribution, among other things. Leaving it commented out was not helping anyone.

Builds on #652 please review and merge that first

### CODE REVIEW

File: cmd/bnsd/app/app_test.go [hunk 1]
```diff
@@ -193,6 +195,65 @@ func sendToken(t *testing.T, baseApp abci.Application, chainID string, height in
...
+		messages = append(messages,
+			app.BatchMsg_Union{
+				Sum: &app.BatchMsg_Union_SendMsg{
+					SendMsg: msg,
+				},
+			})
+	}
+
+	if fail == true {
+		messages[batch.MaxBatchMessages-1] = app.BatchMsg_Union{
```
Comment (alpe) [inline cmd/bnsd/app/app_test.go]: PP: I was confused to find an escrow msg here. Please add a comment why you do this

File: cmd/bnsd/app/app_test.go [hunk 2]
```diff
@@ -193,6 +195,65 @@ func sendToken(t *testing.T, baseApp abci.Application, chainID string, height in
 	return res
 }
 
+// checks batchWorks
+func sendBatch(t *testing.T, fail bool, baseApp weaveApp.BaseApp, chainID string, height int64, signers []Signer,
```
Comment (alpe) [inline cmd/bnsd/app/app_test.go]: Req: this is not called from anywhere

**Comment by ethanfrey** [comment]
Good eye, the code was never called and was indeed broken.

I called it, cleaned it up, and then fixed the broken code paths.
It should all work now. Happy for another review.

**Comment by alpe** [comment]
perfect, looks good now!

**Pull request merged**